Scripts/DCS-BIOS/lib/modules/aircraft_modules/MiG-29A.lua

```diff
@@ -189,7 +190,126 @@ function MiG_29A:defineCabinTempSwitch(identifier, device_id, arg_number, catego
 	end)
 end
 
+local function gunTriggerIntValue(arg_value)
+	if arg_value < 0.30 then
+		return 0
+	elseif arg_value < 0.675 then
+		return 1
+	else
+		return 2
+	end
+end
+
+function MiG_29A:defineGunTrigger(identifier, device_id, arg_number, category, description)
+	local alloc = self:allocateInt(2)
+	self:addExportHook(function(dev0)
+		local val = gunTriggerIntValue(dev0:get_argument_value(arg_number))
+		alloc:setValue(val)
+	end)
+
+	local control = Control:new(category, ControlType.toggle_switch, identifier, description, {
+		SetStateInput:new(2, "set the trigger position"),
+		FixedStepInput:new("switch to previous or next state"),
+	}, {
+		IntegerOutput:new(alloc, Suffix.none, "trigger position -- 0 = released, 1 = first detent , 2 = second detent"),
+	})
+	self:addControl(control)
+
+	self:addInputProcessor(identifier, function(toState)
+		local dev = GetDevice(device_id)
+
+		if dev == nil then
+			return
+		end
+
+		local currentState = gunTriggerIntValue(GetDevice(0):get_argument_value(arg_number))
+		local new_state
+
+		if toState == "INC" then
+			if currentState >= 2 then
+				return
+			end
+			new_state = currentState + 1
+		elseif toState == "DEC" then
+			if currentState <= 0 then
+				return
+			end
+			new_state = currentState - 1
+		else
+			new_state = tonumber(toState)
+		end
+
+		if new_state == 0 then -- RELEASED
+			dev:performClickableAction(3001, 0)
+			dev:performClickableAction(3002, 0)
+		elseif new_state == 1 then -- FIRST DETENT
+			dev:performClickableAction(3001, 0)
+			dev:performClickableAction(3002, 0.6)
+		elseif new_state == 2 then -- SECOND DETENT
+			dev:performClickableAction(3001, 0)
+			dev:performClickableAction(3001, 1)
+		end
+	end)
+end
+
+--- Adds an input-only control which performs a specific LoSetCommand with no arguments
+--- @param identifier string the unique identifier for the control
+--- @param iCommand ICommand the dcs icommand to move the switch up or down
+--- @param arg_number integer the dcs argument number
+--- @param category string the category in which the control should appear
+--- @param description string additional information about the control
+--- @return Control control the control which was added to the module
+function MiG_29A:defineBrakeLever(identifier, iCommand, arg_number, category, description)
```

Comment:
I'd like for us to move this to Module.lua if possible (and include some tests too). Perhaps it would be good to just reserve the space in this pr, and have a separate pr just for adding the icommand axis control?